src/mgmt/srv_query.c

```diff
@@ -811,26 +761,17 @@ ds_mgmt_dev_identify(uuid_t dev_uuid, Ctl__DevIdentifyResp *resp)
 	}
 	uuid_unparse_lower(dev_uuid, resp->dev_uuid);
 
-	D_ALLOC(resp->led_state, buflen);
-	if (resp->led_state == NULL) {
-		D_ERROR("Failed to allocate device led state");
-		rc = -DER_NOMEM;
-		goto out;
-	}
-
 	uuid_copy(identify_info.devid, dev_uuid);
 	rc = dss_ult_execute(bio_storage_dev_identify, &identify_info, NULL,
 			     NULL, DSS_XS_VOS, 0, 0);
 	if (rc != 0)
 		goto out;
 
-	strcpy(resp->led_state, "IDENTIFY");
+	resp->led_state = NVME_DEV_FL_IDENTIFY;
```

Comment:
The "IDENTIFY" state here is simply just returned from the dmg command, it's not an actual device state. In this patch though you are setting it as a device state, which is it actually reset somewhere?

Again not sure how useful this identify state is, as it only lasts for about 1 min

Reply:
as far as I can see the state is never saved, to me given we are returning a state it makes sense for it to be a bit flag in the existing state value

Reply:
I'd agree with Sydney, I'm not sure identify is useful as a device state. It seems like a separate thing. The device could be in any of the other states and have the identify LED on, couldn't it? That's the part that gets confusing for me.

Reply:
I will defer to @NiuYawei as I followed his advice on adding it to the bitset, it makes sense to me to have it as one of the bit flags but I will go with the consensus. The SmdQuery with identify=true in the request returns a device state so it makes sense to me to return the state with the identify flag set, what is your preferred behaviour @sydneyvanda @kjacque ?

Reply:
I am ok with this, as long as the functionality is currently not changed (ie the state is not returned). We can discuss returning an IDENTIFY state as a later effort if we see a need/use-case for it.

Reply:
Is the IDENTIFY bit only set here? I.e. it's never read up from the device? If so I'd lean toward omitting it from the device state.

Reply:
What I have discussed with @sydneyvanda that currently it's only read as part of the device state returned by SmdQuery in the control plane but in a follow-up we will store the state that will be reflected in output of list-devices. So I'm inclined to leave it in.

Reply:
Ah, okay, I'm fine with leaving it then.